src/luxonis_ml/embeddings/emb_mistakes.py

```diff
@@ -0,0 +1,140 @@
+"""
+Mismatch Detection in Labelled Data
+
+This module provides functionalities to detect mismatches or potential mislabelling 
+in a dataset based on various strategies. This is crucial in supervised machine 
+learning tasks where the quality of labels significantly affects model performance.
+
+Methods implemented:
+    - Centroids: This method identifies mismatches by comparing the distance of data points 
+                 to the centroid of their own class against the distances to centroids of 
+                 other classes.
+    - KNN (k-Nearest Neighbors): This approach leverages the idea that if the majority of data 
+                                 is correctly labelled, then mislabelled data will be corrected 
+                                 by its nearest neighbors.
+    - [Note: DBSCAN was considered but not implemented due to underperformance.]
+
+Usage:
+    To use this module, import the desired methods and provide the embeddings and labels:
+    ```python
+    from mismatch_detection import find_mismatches_centroids, find_mismatches_knn
+
+    # Detect mismatches using centroids
+    mismatches, new_labels = find_mismatches_centroids(X_train, y_train)
+    
+    # Detect mismatches using KNN
+    mismatches, new_labels = find_mismatches_knn(X_train, y_train)
+    ```
+"""
+
+import numpy as np
+
+import scipy.spatial.distance as distance
+from sklearn.neighbors import KNeighborsClassifier
+
+from luxonis_ml.embeddings.qdrant_utils import *
+
+
+def find_mismatches_centroids(X, y):
+    """
+    Find mismatches in the dataset. 
+    A mismatch is defined as a sample that is closer to another centroid than to its own centroid.
+
+    Parameters
+    ----------
+    X : np.array
+        The embeddings to use.
+    y : np.array
+        The targets to use.
+
+    Returns
+    -------
+    np.array
+        The indices of the mismatches.
+    np.array
+        The new predicted labels.
+    """
+    unique_labels = np.unique(y)
+    # Create a mapping from string labels to integer indices
+    label_to_index = {label: index for index, label in enumerate(unique_labels)}
+
+    # calculate centroids of each class
+    centroids = []
+    for label in unique_labels:
+        # Get the indices of the samples in the train set that belong to the specified class
+        indices = np.where(y == label)[0]
+
+        # Get the embeddings of the samples in the specified class
+        embeddings_class = X[indices]
+
+        # Calculate the centroid of the specified class
+        centroid = np.mean(embeddings_class, axis=0)
+        centroids.append(centroid)
+
+    # append the centroids to the embeddings
+    Centroids = np.array(centroids)
```

Comment:
This variable being capitalized makes me think it's a special class, but not a big deal